lib/hci-socket/smp.js

```diff
@@ -28,14 +29,21 @@ var Smp = function(aclStream, localAddressType, localAddress, remoteAddressType,
 
   this._aclStream.on('data', this.onAclStreamDataBinded);
   this._aclStream.on('end', this.onAclStreamEndBinded);
+
+  this._remoteAddress = remoteAddress;
+
+  if(passkeyProvider)
+    this._passkeyProvider = passkeyProvider;
+  else
+    this._passkeyProvider = Promise.resolve;
 };
 
 util.inherits(Smp, events.EventEmitter);
 
 Smp.prototype.sendPairingRequest = function() {
   this._preq = new Buffer([
     SMP_PAIRING_REQUEST,
-    0x03, // IO capability: NoInputNoOutput
+    0x04, // IO capability: KeyboardDisplay
```

Comment:
Did this have to change in order for a passkey to be allowed?

Reply:
Yes